src/sardana/taurus/core/tango/sardana/macroserver.py

```diff
@@ -326,8 +342,12 @@ def __init__(self, name, **kw):
             # TODO: For Taurus 4 compatibility
             from taurus.core import TaurusDevState
             self._old_sw_door_state = TaurusDevState.Undefined
+        try:
+            self.stateObj.addListener(self.stateChanged)
+        except:
```

Comment:
Again the same thing as above. I'm not sure what error would happen here, presumably also `AttributeError`.